alicloud/resource_alicloud_cen_route_entry_publishment.go

```diff
@@ -0,0 +1,163 @@
+package alicloud
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cbn"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAlicloudCenRouteEntryPublishment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCenRouteEntryPublishmentCreate,
+		Read:   resourceAlicloudCenRouteEntryPublishmentRead,
+		Delete: resourceAlicloudCenRouteEntryPublishmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"child_instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"region_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"route_table_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"cidr_block": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCenRouteEntryPublishmentCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	para := convertResourceDataToParas(d)
+	request := cbn.CreatePublishRouteEntriesRequest()
+	request.CenId = para.cenId
+	request.ChildInstanceId = para.childInstanceId
+	instanceType, err := getCenInstanceType(para.childInstanceId)
+	if err != nil {
+		return err
+	}
+	request.ChildInstanceType = instanceType
+	request.ChildInstanceRegionId = para.regionId
+	request.ChildInstanceRouteTableId = para.vtbId
+	request.DestinationCidrBlock = para.cidr
+
+	err = resource.Retry(3*time.Minute, func() *resource.RetryError {
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Building CreatePublishRouteEntriesRequest got an error: %#v", err))
+		}
+		_, err = client.cenconn.PublishRouteEntries(request)
+		if err != nil {
+			if IsExceptedError(err, OperationBlocking) {
+				return resource.RetryableError(fmt.Errorf("Create CEN route entry publish timeout and got an error: %#v.", err))
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Create CEN route entry publish timeout, CEN %s child instance %s vtb %s cidr %s, error info: %#v.", para.cenId, para.childInstanceId, para.vtbId, para.cidr, err)
+	}
+
+	d.SetId(para.vtbId + COLON_SEPARATED + para.cidr)
```

Comment:
I think id should contain `cen_id`. If there are two cen instances to publish the same route entry, it will not work.